src/osd/osd_types.h

```diff
@@ -3757,8 +3757,9 @@ class pg_missing_set : public pg_missing_const_i {
     } else if (e.is_delete()) {
       rm(e.soid, e.version);
     }
-
-    tracker.changed(e.soid);
+    if (!e.is_error()) {
```

Comment:
now that I look the only 2 callers of this already avoid adding ERROR entries, so this isn't needed